### ISSUE TITLE
Sealed interfaces should allow implementations of it's interfaces

### ISSUE THREAD

**Issue by oprypkhantc** [jiripudil/phpstan-sealed-classes#3]
Hey.

Currently, if you try to seal a set of interfaces (without implementations), you get an error when trying to actually implement one of the interfaces:

```php
#[Sealed(permits: [ClassReflection::class, EnumReflection::class])]
interface TypeReflection {}

interface ClassReflection implements TypeReflection {}
interface EnumReflection implements TypeReflection {}

// Type SomeClassReflection is not allowed to be a subtype of TypeReflection
class SomeClassReflection implements ClassReflection {}
```

This doesn't quite make sense because any inheritors of `ClassReflection` and `EnumReflection` will still be instances of `ClassReflection` or `EnumReflection` and will not break any contracts or `instanceof` checks.

You've referred to Kotlin multiple times here, so I'll refer to Kotlin too. I've checked how sealed interfaces work there and it follows that same logic:

```kotlin
// test/nested/Test.kt
package test.nested

sealed interface TypeReflection

interface ClassReflection : TypeReflection
interface EnumReflection : TypeReflection

// testtwo/nestedtwo/Test.kt
package testtwo.nestedtwo

import test.nested.ClassReflection

class SomeClassReflection : ClassReflection // <-- no error reported here

```

Would you accept a PR changing this behaviour to match Kotlin's?

**Comment by jiripudil** [comment]
Hi, thanks for the report! The behaviour you propose is correct and has, in fact, been implemented here from the start – along with [a test](https://github.com/jiripudil/phpstan-sealed-classes/blob/15687cda0cb106f38aed5d3666db1ad7b9305b11/tests/data/sealed-interface.php#L14). So I suspect there might be something else at play if you're getting this error.

Could you please provide a reproducible example that I could investigate? A PR with a failing test would be much appreciated.

**Comment by oprypkhantc** [comment]
Closing this to avoid having a stale issue. Unfortunately I wasn't able to reproduce this as I've only seen this while working on the code, and by the time I got to commit it - it wasn't there there anymore. My best guess is that I did something wrong :)
